client/client.go

```diff
@@ -1576,6 +1576,8 @@ func (c *Client) setupNode() error {
 				}
 				node.HostVolumes[k] = v.Copy()
 			}
+		} else {
+			node.HostVolumes = make(map[string]*structs.ClientHostVolumeConfig, 0)
```

Comment:
The other branch of this makes the map based on the length, so we could save ourselves a conditional by having this whole thing be:

```go
if node.HostVolumes == nil {
	node.HostVolumes = make(map[string]*structs.ClientHostVolumeConfig, len(newConfig.HostVolumes))
	for k, v := range newConfig.HostVolumes {
		if _, err := os.Stat(v.Path); err != nil {
			return fmt.Errorf("failed to validate volume %s, err: %w", v.Name, err)
		}
		node.HostVolumes[k] = v.Copy()
	}
}
```

Reply:
5feb3b5